From 5fb17312500bcad280f4edc2828546170788209f Mon Sep 17 00:00:00 2001
From: "Darryl L. Pierce" <dpierce@redhat.com>
Date: Fri, 21 Sep 2012 14:47:58 -0400
Subject: [PATCH] Provides systemd support to qpidd.

Systemd is a replacement for SysVInit scripts for starting up daemons on
Linux systems.
---
 qpid/cpp/etc/Makefile.am      |  8 +++++---
 qpid/cpp/etc/qpidd.service.in | 13 +++++++++++++
 2 files changed, 18 insertions(+), 3 deletions(-)
 create mode 100644 qpid/cpp/etc/qpidd.service.in

diff --git a/qpid/cpp/etc/Makefile.am b/qpid/cpp/etc/Makefile.am
index aa41c65..7af495f 100644
--- a/qpid/cpp/etc/Makefile.am
+++ b/qpid/cpp/etc/Makefile.am
@@ -20,7 +20,7 @@ SASL_CONF = sasl2/qpidd.conf
 
 EXTRA_DIST = \
 	$(SASL_CONF) \
-	qpidd.in qpidd-primary.in qpidd.conf qpidc.conf CMakeLists.txt \
+	qpidd.service.in qpidd.in qpidd-primary.in qpidd.conf qpidc.conf CMakeLists.txt \
 	cluster.conf-example.xml.in
 
 confdir = $(sysconfdir)/qpid
@@ -47,9 +47,11 @@ qpidd-primary: qpidd-primary.in
 	sed $(SUBST) $< > $@
 cluster.conf-example.xml: cluster.conf-example.xml.in
 	sed $(SUBST) $< > $@
+qpidd.service: qpidd.service.in
+	sed $(SUBST) $< > $@
 
-CLEANFILES = qpidd qpidd-primary cluster.conf-example.xml
+CLEANFILES = qpidd qpidd-primary cluster.conf-example.xml qpidd.service
 
 initddir = $(sysconfdir)/init.d
-nobase_initd_SCRIPTS = qpidd qpidd-primary
+nobase_initd_SCRIPTS = qpidd qpidd-primary qpidd.service
 
diff --git a/qpid/cpp/etc/qpidd.service.in b/qpid/cpp/etc/qpidd.service.in
new file mode 100644
index 0000000..ec89e12
--- /dev/null
+++ b/qpid/cpp/etc/qpidd.service.in
@@ -0,0 +1,13 @@
+[Unit]
+Description=An AMQP message broker daemon.
+Documentation=man:qpidd(1) http://qpid.apache.org/
+After=network.target
+
+[Service]
+User=qpidd
+Group=qpidd
+Type=simple
+ExecStart=/usr/sbin/qpidd --config /etc/qpidd.conf
+
+[Install]
+WantedBy=multi-user.target
-- 
1.7.11.4

